mssmt/store.go

```diff
@@ -244,29 +252,66 @@ func (d *DefaultStore) DeleteCompactedLeaf(key NodeHash) error {
 	return nil
 }
 
+// DeleteRoot deletes the root node of the MS-SMT.
+func (d *DefaultStore) DeleteRoot() error {
+	d.root = nil
+	d.cntDeletes++
+
+	return nil
+}
+
+// DeleteAllNodes deletes all nodes in the MS-SMT.
+func (d *DefaultStore) DeleteAllNodes() error {
+	// Delete leaves, then compacted leaves, then branches.
+	leafCount := len(d.leaves)
+	maps.Clear(d.leaves)
+	d.cntDeletes += leafCount
+
+	compactedLeafCount := len(d.compactedLeaves)
+	maps.Clear(d.compactedLeaves)
+	d.cntDeletes += compactedLeafCount
+
+	branchCount := len(d.branches)
+	maps.Clear(d.branches)
+	d.cntDeletes += branchCount
+
+	return nil
+}
+
 // GetChildren returns the left and right child of the node keyed by the given
 // NodeHash.
 func (d *DefaultStore) GetChildren(height int, key NodeHash) (
 	Node, Node, error) {
 
+	EmptyNodeHash := EmptyTree[height].NodeHash()
```

Comment:
This should be a lower case var, otherwise it looks like a package level exported constant/var.